### DISCUSSION TITLE
Separate password (or delay) to access options of a specific block set

### DISCUSSION THREAD

**Discussion by Rodvi** [proginosko/LeechBlockNG#454] [Ideas]
Currently, the only way to protect options for a specific block set is to enable the "Prevent access to options for this block set at times when these sites are blocked" option. Unfortunately, this option doesn't work well in many cases, for example in the following case. 

I have a block set of sites that is designed to be active forever (I think that most of LeechBlock users have their own almost-forever-active block set too). But very rarely (maybe once a month) I need to remove a site from this list or change its options. Besides this rare scenario, I don't want to have access to change options of this block set (note that this does not refer to options of other block sets). So I did the following - made this block set active every day in 0000-2358 and checked "Prevent access to options for this block set at times when these sites are blocked". So when I need to change options for this block set, I wait until 2358 and then change them. This approach has two major drawbacks. The first is that if I want to make a change in the morning, I have to wait 12-16 hours until 2358 (of course, I can create more windows in the schedule, like 0000-0958,1000-1658,1700-2358; but this will significantly reduce the protection because I will be able to deactivate the block set during each of these 2min periods). The second disadvantage is that first I have to click on LeechBlock > `Options` and thus I get access to options of all block sets and General options (there is no direct button "Change options of a specific block set" which will prevent access to all other block sets and General options). 

The perfect solution should be the following: LeechBlock should provide user a possibility to make a separate password (or delay that is better in most cases) for access to options of a specific block set. Ideally, there should be a button `Options of a blockset` (available without going through the ordinary options page) that will prompt the user to select a block set name and then enter its password (or wait for a predefined delay) and after that the separate window should open where there will be only the options of a selected block set (there should be no access to options of other block set or General options).

**Comment by namankumar9** [comment]
I agree with all the points. There should be a password option too to prevent Edits to a Block Set apart from that "Prevent access to options for this block set at times when these sites are blocked" option.
